src/condor_utils/ipv6_interface.cpp

```diff
@@ -42,6 +42,59 @@ uint32_t find_scope_id(const condor_sockaddr& addr) {
 	return result;
 }
 
+
+#else
+	// Win32
+
+#include <iphlpapi.h>
+
+uint32_t find_scope_id(const condor_sockaddr& addr) {
+	if (!addr.is_ipv6())
+		return 0;
+
+	ULONG family = AF_INET6;
+	ULONG flags = GAA_FLAG_INCLUDE_PREFIX; // Set the flags to pass to GetAdaptersAddresses
+	ULONG cbBuf = 16*1024; // start with 16Kb
+	PIP_ADAPTER_ADDRESSES pAddresses = nullptr;
+	auto_free_ptr addrbuf((char*)malloc(cbBuf));
+	int iterations = 0;
+
+	char ipbuf[IP_STRING_BUF_SIZE+1];
+	dprintf(D_ALWAYS, "Win32 find_scope_id(%s) id called\n", addr.to_ip_string(ipbuf, IP_STRING_BUF_SIZE, false));
+
+	do {
+		ASSERT(addrbuf.ptr())
+		pAddresses = (PIP_ADAPTER_ADDRESSES)addrbuf.ptr();
+		DWORD retval = GetAdaptersAddresses(family, flags, NULL, pAddresses, &cbBuf);
+		if (retval == ERROR_BUFFER_OVERFLOW) {
+			addrbuf.set((char*)malloc(cbBuf));
+			continue;
+		} else if (retval == NO_ERROR) {
+			break;
+		} else {
+			dprintf(D_ERROR, "ipv6_get_scope_id() error %u from GetAdaptersAddresses\n", retval);
+			return 0;
+		}
+	} while (++iterations < 3);
```

Comment:
The code walks `pAddresses` after this even if the buffer overflowed.  I assume that the buffer-overflow error indicates a partial copy into the buffer, but even so, this seems likely to cause problems at some point.